src/app/compiler.js

```diff
@@ -173,12 +173,14 @@ function Compiler (editor, queryParams, handleGithubCall, updateFiles) {
           break;
         case 'compiled':
           var result;
+          var source;
           try {
             result = JSON.parse(data.data);
+            source = JSON.parse(data.source);
```

Comment:
I would prefer to have two have two separate try/catch blocks for this. The `source` one should never trigger.